src/python/grpcio/grpc/experimental/aio/_base_call.py

```diff
@@ -158,6 +158,23 @@ def __aiter__(self) -> AsyncIterable[ResponseType]:
           stream.
         """
 
+    @abstractmethod
+    async def try_connect(self) -> None:
+        """Tries to connect to peer and raise aio.AioRpcError if failed.
+
+        This is an EXPERIMENTAL method.
+
+        This method is available for streaming RPCs. This method enables the
```

Comment:
It seems odd that this method would exist on three out of four arities. For symmetry's sake, couldn't this method also be added to unary RPCs?

Reply:
For unary calls, everything happen in one batch, which is beneficial to our performance. Unless we split it into two batch in Core (send and receive), there is not way to cleanly `try_connect`.

Another alternative is to cheat a little bit, make the `try_connect` equivalent to `__await__` which finishes the RPC.

I don't like both options for now, so just added the new API for 3 streaming arities.

Reply:
I was thinking of the second option (the first one is also valid, but my suspicion is that it would adversely affect performance). I think `try_connect` completing the RPC is fine. In some cases, it may also complete a streaming RPC. But the symmetry means that you can write wrapper code that works regardless of the arity. No conditionals required.